src/expressionEvaluator/expressionEvaluator.ts

```diff
@@ -0,0 +1,30 @@
+import { IExpressionEvaluator } from './iExpressionEvaluator'
+import { ExpressionNodeEvaluator } from './expressionNodeEvaluator'
+import { ExpressionNodeType } from './expressionNodeType'
+import { ExpressionFunction } from './expressionFunction'
+import { ExpressionContext } from './expressionContext'
+import { JSExpr } from './jsExpr'
+import jsep from './jsep'
+
+export class ExpressionEvaluator implements IExpressionEvaluator {
+  functions: { [functionName: string]: ExpressionFunction }
+  evaluators: { [nodeType in ExpressionNodeType]: ExpressionNodeEvaluator }
+
+  parseExpression(expressionString: string): JSExpr {
+    return jsep(expressionString)
+  }
+
+  evaluateString(expression: string, context: ExpressionContext): any {
```

Comment:
should the context be set at constructor level?
So we won't need to pass the context to every node evaluator ?

Reply:
in some cases (member expression node evaluation) the context changes, for example when a member expression node in a record expression will be evaluated, the context "record node" can change.
e.g. tree[1].tree_id : the initial context node is "plot", but tree[1] gives you a tree node, that will be the context node to evaluate tree_id...